### PR TITLE
Add filter flags for Audit & Scan commands

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
- Added `min-severity` flag to filter scan/audit results with lower severity than the `min-severity`. 
- Added `fixable-only` flag to filter scan/audit results without a fixing version.

### CODE REVIEW

File: xray/audit/commonutils.go [hunk 1]
```diff
@@ -100,7 +99,7 @@ func GetAndAssertNode(t *testing.T, modules []*services.GraphNode, moduleId stri
 	return module
 }
 
-// Get a specific module from the provided modules list
+// GetModule Get a specific module from the provided modules list
```
Comment (eyalbe4) [inline xray/audit/commonutils.go:102]: ... Get a specific... --> ... gets a specific...

File: xray/audit/commonutils.go [hunk 2]
```diff
@@ -115,7 +114,7 @@ func GetModule(modules []*services.GraphNode, moduleId string) *services.GraphNo
 	return nil
 }
 
-// Gets executable version and prints to the debug log if possible.
+// GetExecutableVersion Gets an executable version and prints to the debug log if possible.
```
Comment (eyalbe4) [inline xray/audit/commonutils.go:117]: Gets --> gets

File: xray/commands/audit/generic/auditmanager.go [hunk 3]
```diff
@@ -26,22 +26,24 @@ import (
...
-	xrayVersion         string
+	xrayGraphScanParams  *services.XrayGraphScanParams
+	serverDetails        *config.ServerDetails
+	progress             ioUtils.ProgressMgr
+	dependencyTrees      []*services.GraphNode
+	ignoreConfigFile     bool
+	excludeTestDeps      bool
+	insecureTls          bool
+	useWrapper           bool
+	withFixVersionFilter bool
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:37]: withFixVersionFilter --> filterIssuesWithoutFix
Let's also change the related functions and variables names.

File: xray/commands/audit/generic/auditmanager.go [hunk 4]
```diff
@@ -26,22 +26,24 @@ import (
...
+	minSeverityFilter    string
```
Comment (eyalbe4) [inline xray/commands/audit/generic/auditmanager.go:38]: minSeverityFilter --> filterBelowMinSeverity
Let's also change the related functions and variables names.

Let's also add a description to this param, describing its allowed values.

File: xray/commands/utils.go [hunk 5]
```diff
@@ -14,6 +15,84 @@ const (
 	BypassArchiveLimitsMinXrayVersion = "3.59.0"
 )
 
+type FilterLevel int
```
Comment (eyalbe4) [inline xray/commands/utils.go:18]: FilterLevel --> SeverityLevel
Let's also change the related functions.

File: xray/commands/utils.go [hunk 6]
```diff
@@ -14,6 +15,84 @@ const (
...
+
+const (
+	NoFilter FilterLevel = 0
+	Low      FilterLevel = 1
+	Medium   FilterLevel = 2
+	High     FilterLevel = 3
+	Critical FilterLevel = 4
+)
+
+var mapFilterNameToLevel = map[string]FilterLevel{
```
Comment (eyalbe4) [inline xray/commands/utils.go:28]: mapFilterNameToLevel --> nameToFilter

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: xray/audit/commonutils.go (hunk 1 above)
Comment (omerzi) [reply xray/audit/commonutils.go:102]: done

File: xray/audit/commonutils.go (hunk 2 above)
Comment (omerzi) [reply xray/audit/commonutils.go:117]: done

File: xray/commands/utils.go (hunk 5 above)
Comment (omerzi) [reply xray/commands/utils.go:18]: done

File: xray/commands/utils.go (hunk 6 above)
Comment (omerzi) [reply xray/commands/utils.go:28]: done

File: xray/commands/utils.go [hunk 7]
```diff
@@ -14,6 +15,84 @@ const (
 	BypassArchiveLimitsMinXrayVersion = "3.59.0"
 )
 
+type SeverityLevel int
+
+const (
```
Comment (sverdlov93) [inline xray/commands/utils.go:20]: the following logic already exists.
You can use the 'numValue' here:
var severities = map[string]*severity{
	"Critical": {emoji: "💀", title: "Critical", numValue: 4, style: color.New(color.BgLightRed, color.LightWhite)},
	"High":     {emoji: "🔥", title: "High", numValue: 3, style: color.New(color.Red)},
	"Medium":   {emoji: "🎃", title: "Medium", numValue: 2, style: color.New(color.Yellow)},
	"Low":      {emoji: "👻", title: "Low", numValue: 1},
}

**Pull request merged**